hail/src/main/scala/is/hail/expr/ir/BlockMatrixIR.scala

```diff
@@ -344,6 +344,39 @@ case class BlockMatrixBroadcast(
   }
 }
 
+case class BlockMatrixAgg(
+  child: BlockMatrixIR,
+  outIndexExpr: IndexedSeq[Int],
+  blockSize: Int,
```

Comment:
why doesn't this just come from `child.typ.blockSize`?

Reply:
Wasn't sure if after aggregating you would want to adjust the block size. I guess either way that's not really a concern of IR at this level.

Reply:
Hmm, my intuition is that this should be a separate IR node to reduce block size. I think we should leave it out for now and add it when we have a better idea of how it will work. We don't currently support reducing the block size, right?

Reply:
Nope, and that makes sense. Will cut it out.